src/mame/heathkit/h89.cpp

```diff
@@ -300,7 +307,53 @@ void h89_state::h89_io(address_map &map)
 	map(0xfa, 0xfb).rw(FUNC(h89_state::raise_NMI_r), FUNC(h89_state::raise_NMI_w));
 }
 
+void h89_state::h89_io(address_map &map)
+{
+	h89_base_io(map);
+
+	// Disk I/O #1 - 0170-0173 (0x78-0x7b)
+	//   Options
+	//     - H37 5-1/4" Soft-sectored Controller - Requires MTR-90 ROM
+	//     - H47 Dual 8" Drives - Requires MTR-89 or MTR-90 ROM
+	//     - H67 8" Hard disk + 8" Floppy Drives - Requires MTR-90 ROM
+	map(0x78, 0x7b).rw(m_h37, FUNC(heath_z37_fdc_device::read), FUNC(heath_z37_fdc_device::write));
+}
+
+void h89_state::h88_io(address_map &map)
+{
+	h89_base_io(map);
+
+	// Cassette I/O (uses 0xf8 - 0xf9) - Requires MTR-88 ROM
+	map(0xf8, 0xf9).rw(m_cassette, FUNC(heath_h_88_cass_device::read), FUNC(heath_h_88_cass_device::write));
+
+	// 5-1/4" Hard-sectored Controller - only other card supported MTR-88.
+	// map(0x7c, 0x7f)
```

Comment:
Please keep things in ascending address order if there isn’t a compelling reason not to.